util/session/sessionmanager.go

```diff
@@ -268,19 +268,26 @@ func (mgr *SessionManager) Parse(tokenString string) (jwt.Claims, string, error)
 		return nil, "", fmt.Errorf("account password has changed since token issued")
 	}
 
+	exp, err := jwtutil.ExpirationTime(claims)
+	if err != nil {
+		return nil, "", fmt.Errorf("token has no expiration time: %s", err)
+	}
+
 	newToken := ""
-	if exp, err := jwtutil.ExpirationTime(claims); err == nil {
-		tokenExpDuration := exp.Sub(issuedAt)
-		remainingDuration := time.Until(exp)
+	remainingDuration := time.Until(exp)
+	if remainingDuration > 0 && remainingDuration < autoRegenerateTokenDuration && capability == settings.AccountCapabilityLogin {
```

Comment:
I don't follow your contention that if `remainingDuration` is <= 0 it would result in a token with no expiration. The `secondsBeforeExpiry` value passed to the `Create` function is computed from the difference between the current tokens expiration time and its issuedAt time, which aren't dependent on the current time or the value of `remainingDuration`.

I'm not saying that this check shouldn't necessarily be here, just that it seems like there must be some other rationale.

Reply:
Wait, I see what you're saying: if you don't enforce token expiration here, this code will generate a newToken regardless of how old the input token is. So not literally that the `Create` function will generate a token that lacks an `ExpiresAt` claim (which it would if you passed a <= 0 `secondsBeforeExpiry` value).

That definitely seems bad if the existing code could be used to indefinitely re-up a login session.